akka-http-core/src/main/scala/akka/http/scaladsl/model/headers/HttpChallenge.scala

```diff
@@ -14,8 +14,15 @@ final case class HttpChallenge(scheme: String, realm: String,
 
   def render[R <: Rendering](r: R): r.type = {
     r ~~ scheme
-    if (realm != null) r ~~ " realm=" ~~#! realm
-    if (params.nonEmpty) params.foreach { case (k, v) ⇒ r ~~ ',' ~~ k ~~ '=' ~~# v }
+
+    if (realm != null) {
+      r ~~ " realm=" ~~#! realm
+      params.foreach { case (k, v) ⇒ r ~~ ',' ~~ k ~~ '=' ~~# v }
```

Comment:
I'm not 100% sure what you mean, but I don't think it will pass. You also have to get the whitespace between scheme and attributes right. I tried several variants, but this seems to be the easiest way to do it. Maybe I missed something...